tests/network/vmi_infosource.go

```diff
@@ -0,0 +1,146 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8snetworkplumbingwgv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kvirtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+var _ = SIGDescribe("Infosource", func() {
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		Expect(err).NotTo(HaveOccurred(), "Should successfully initialize an API client")
+
+		tests.BeforeTestCleanup()
+	})
+
+	Context("with a vmi and a bridge interface", func() {
+		var vmi *kvirtv1.VirtualMachineInstance
+
+		const defaultInterfaceName = "default"
+		const originalMac = "02:00:05:05:05:05"
+		const updatedMac = "02:00:b5:b5:b5:b5"
+
+		secondaryLinuxBridgeInterface := kvirtv1.Interface{
+			Name: "secondary",
+			InterfaceBindingMethod: kvirtv1.InterfaceBindingMethod{
+				Bridge: &kvirtv1.InterfaceBridge{},
+			},
+			MacAddress: "02:00:a0:a0:a0:a0",
+		}
+
+		linuxBridgeNetwork := kvirtv1.Network{
+			Name: secondaryLinuxBridgeInterface.Name,
+			NetworkSource: kvirtv1.NetworkSource{
+				Multus: &kvirtv1.MultusNetwork{
+					NetworkName: secondaryLinuxBridgeInterface.Name,
+				},
+			},
+		}
+
+		BeforeEach(func() {
+			By("Create NetworkAttachmentDefinition")
+			nad := generateNetworkAttachmentDefinition()
+			_, err := virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(util.NamespaceTestDefault).Create(context.TODO(), nad, metav1.CreateOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
```

Comment:
@orelmisan , you have raised interest to use something like this in the multus tests. It seems this package was already used, so maybe it will fit to what you need as well.